lib/to-interpolated-regexp.js

```diff
@@ -3,7 +3,7 @@ module.exports = function(source) {
     var splitSource = source.split('{componentName}');
     return function(componentName) {
       try {
-        return new RegExp(splitSource.length > 1 ? splitSource[0] + componentName + splitSource[1] : splitSource[0]);
+        return new RegExp(splitSource.join(componentName));
```

Comment:
@davidtheclark 
I just pushed a test case which uses two custom configs, one which uses the `{componentName}` interpolation and one which doesn't.